File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Admin/AdminAuditLogDTO.cs
using platform_core_service.Common.Entities.BaseEntity;
using platform_core_service.Common.Utils.Enums;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Admin
{
    public class AdminAuditLogDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public string? ActorId { get; set; }
        public string? ActorUserId { get; set; }
        public string? ActorDisplayName { get; set; }
        public string? ActorRole { get; set; }
        public AuditTargetType TargetType { get; set; }
        public string TargetId { get; set; } = null!;
        public string? TargetDisplayName { get; set; }
        public AuditActionType ActionType { get; set; }
        public string? OldState { get; set; }
        public string? NewState { get; set; }
        public string? PublicReason { get; set; }
        public string? InternalNote { get; set; }
        public string? MetadataJson { get; set; }
        public string? IpAddress { get; set; }
        public string? UserAgent { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
    }
}

File: frontend/src/types/admin/admin-audit-log-dto.ts
export enum AuditTargetType {
  User = 0,
  Post = 1,
  ModerationQueueEntry = 2,
  ModerationReport = 3,
}

export enum AuditActionType {
  UserSuspended = 0,
  UserUnsuspended = 1,
  UserPermanentlyBanned = 2,
  UserRoleChanged = 3,

  PostForceApproved = 10,
  PostForceRejected = 11,

  ModerationQueueApproved = 20,
  ModerationQueueRejected = 21,

  ReportAssigned = 30,
  ReportResolved = 31,
  ReportDismissed = 32,
  ReportEscalated = 33,
  ReportTargetActionExecuted = 34,
}

export interface AdminAuditLogDTO {
  id: string
  actorId?: string | null
  actorUserId?: string | null
  actorDisplayName?: string | null
  actorRole?: string | null
  targetType: AuditTargetType | number
  targetId: string
  targetDisplayName?: string | null
  actionType: AuditActionType | number
  oldState?: string | null
  newState?: string | null
  publicReason?: string | null
  internalNote?: string | null
  metadataJson?: string | null
  ipAddress?: string | null
  userAgent?: string | null
  createdAt: string
}
